addons/payment_stripe/static/src/js/stripe.js

```diff
@@ -41,27 +41,46 @@ odoo.define('payment_stripe.stripe', function(require) {
             $(this).attr('disabled','disabled');
 
         var $form = $(e.currentTarget).parents('form');
-        var acquirer_id = $(e.currentTarget).closest('div.oe_sale_acquirer_button,div.oe_quote_acquirer_button').data('id');
+        var acquirer_id = $(e.currentTarget).closest('div.oe_sale_acquirer_button,div.oe_quote_acquirer_button,div.o_website_payment_new_payment');
+        acquirer_id = acquirer_id.data('id') || acquirer_id.data('acquirer_id');
         if (! acquirer_id) {
             return false;
         }
 
         var so_token = $("input[name='token']").val();
-        var so_id = $("input[name='return_url']").val().match(/quote\/([0-9]+)/);
+        var so_id = $("input[name='return_url']").val().match(/quote\/([0-9]+)/) || undefined;
         if (so_id) {
             so_id = parseInt(so_id[1]);
         }
 
         e.preventDefault();
-        ajax.jsonRpc('/shop/payment/transaction/' + acquirer_id, 'call', {'so_id': so_id, 'so_token': so_token}).then(function (data) {
-            $form.html(data);
-            handler.open({
-                name: $("input[name='merchant']").val(),
-                description: $("input[name='invoice_num']").val(),
-                currency: $("input[name='currency']").val(),
-                amount: $("input[name='amount']").val()*100
+        if ($('.o_website_payment')) {
+            ajax.jsonRpc('/website_payment/transaction', 'call', {
+                    reference: $("input[name='invoice_num']").val(),
```

Comment:
Hum your are pointing to `website_quote`, which is not necessarily installed. Moreover, the method clearly needs an `order_id`, while here we are paying an invoice. Have a look at https://github.com/odoo-dev/odoo/blob/saas-11-opw-693421-stripe-nim/addons/website_payment/static/src/js/payment_transaction.js#L21. The comment seems to match what we are doing:
`When clicking on payment button: create the tx using json then continue to the acquirer`

The reference is fine BTW, it gets the invoice number. If you try to pay the same invoice several time, it also adds a unique id at the end of the invoice number.